Code/MinimalLib/minilib.cpp

```diff
@@ -110,10 +110,10 @@ std::string JSMol::get_svg_with_highlights(const std::string &details) const {
   return MinimalLib::mol_to_svg(*d_mol, w, h, details);
 }
 
-std::string JSMol::get_inchi() const {
+std::string JSMol::get_inchi(const std::string &options) const {
   assert(d_mol);
   ExtraInchiReturnValues rv;
-  return MolToInchi(*d_mol, rv);
+  return MolToInchi(*d_mol, rv, options.c_str());
```

Comment:
```suggestion
  return MolToInchi(*d_mol, rv, !options.empty() ? options.c_str() : nullptr);
```

Reply:
``cffiwrapper`` already supports options but in json format.
 
```javascript
  inchi = get_inchi(pkl, pkl_size, "{\"options\":\"/FixedH\"}");
```
MolToInchi seems to accept options as string ``"-FixedH"`` https://github.com/rdkit/rdkit/blob/6f3275f9274eb958492679cb9c9e4af7cd3cbba3/External/INCHI-API/inchi.h#L105

```javascript
mol.get_inchi("-FixedH")
```

Reply:
Ah, sorry, I had missed this comment. Then I would suggest the following:
Extract the JSON options parsing snippet into a function:
```C++
std::string parse_inchi_options(const char *details_json) {
  std::string options;
  if (details_json && strlen(details_json)) {
    boost::property_tree::ptree pt;
    std::istringstream ss;
    ss.str(details_json);
    boost::property_tree::read_json(ss, pt);
    LPT_OPT_GET(options);
  }
  return options;
}
```
and put the function inside the `MinimalLib` namespace in `common.h`, thus it can be called from CFFI `get_inchi()`, `get_inchi_for_molblock()` (and we get rid the current code duplication in `get_inchi_for_molblock()`, which is a bit ugly), and also from JS MinimalLib `get_inchi()`.
This avoids code duplication and yields a consistent API across JS and CFFI MinimalLib.